Makefile

```diff
@@ -452,11 +452,19 @@ ifeq ($(GOOS),linux)
 
 include tools/istio-docker.mk
 
-endif # end of docker block that's restricted to Linux
+# if first part of URL (i.e., hostname) is gcr.io then upload istioctl and deb
+$(if $(findstring gcr.io,$(firstword $(subst /, ,$(HUB)))),$(eval push: gcs.push.istioctl-all gcs.push.deb),)
+
+push: docker.push installgen
 
 gcs.push.istioctl-all: istioctl-all
 	gsutil -m cp -r "${ISTIO_OUT}"/istioctl-* "gs://${GS_BUCKET}/pilot/${TAG}/artifacts/istioctl"
 
+gcs.push.deb: deb
+	gsutil -m cp -r "${ISTIO_OUT}"/*.deb "gs://${GS_BUCKET}/pilot/${TAG}/artifacts/debs"
```

Comment:
Given my prior comment, I now see that the original bin/push script placed the debian files under different directories ("pilot" vs "auth").  On the one hand it seems inappropriate to place the monolithic debian under "pilot", but on the other hand both copies are placing the artifacts in sibling directories and there's a certain appeal to having the binaries close to each other.  Maybe neither should be using "pilot", but I don't know how many other things would also need to be updated to handle the rename.  Anyway, I don't have a particular update in mind but maybe others have ideas on what, if anything, should be done.